src/bp/sdk/botpress.d.ts

```diff
@@ -1360,12 +1361,70 @@ declare module 'botpress/sdk' {
     node: string
   }
 
+  export type FormDataField = any
+
   export interface FormData {
     id?: string
     contentType?: string
-    [key: string]: FormData[] | any
+    [key: string]: FormDataField
   }
 
+  interface FormOption {
+    value: string
+    label: string
+    related?: FormField
+  }
+
+  interface FormContextMenu {
+    type: string
+    label: string
+  }
+
+  export interface FormDynamicOptions {
+    /** An enpoint to call to get the options */
+    endpoint: string
+    /** Used with _.get() on the data returned by api to get to the list of items */
+    path?: string
+    /** Field from DB to map as the value of the options */
+    valueField: string
+    /** Field from DB to map as the label of the options */
+    labelField: string
+  }
+
+  export interface FormAdvancedSetting {
+    key: string
+    label: string
+    type: string
+    defaultValue?: FormDataField
+    moreInfo?: FormMoreInfo
+  }
```

Comment:
Shouldn't AdvancedSettings have the same features as normal fields ?  Also, the signature could be shared between them for shared properties